File: Medium/House-Robber.js
function rob(nums) {
  let rob1 = 0,
    rob2 = 0;

  for (let n of nums) {
    let temp = Math.max(n + rob1, rob2);
    rob1 = rob2;
    rob2 = temp;
  }

  return rob2;
}

console.log(rob([1, 2, 3, 1]));

File: Rust/Dynamic Programming/House-Robber.rs
pub struct Solution;

impl Solution {
    pub fn rob(nums: Vec<i32>) -> i32 {
        let (mut rob1, mut rob2) = (0, 0);
        for num in nums {
            let temp = rob2.max(rob1 + num);
            rob1 = rob2;
            rob2 = temp;
        }
        rob2
    }
}

fn main() {}
